app/policies/wiki_policy.rb

```diff
@@ -35,14 +35,35 @@ def destroy?
     user.admin?
   end
 
-  class Scope < Scope
-    def resolve
-      # if user.present?
-      #   scope.all
-      # else
-      #   scope.where(private: false)
-      # end
-      scope.all
-    end
-  end
+class Scope
+   attr_reader :user, :scope
+
+   def initialize(user, scope)
+     @user = user
+     @scope = scope
+   end
+
+   def resolve
+     wikis = []
+     if user.role == 'admin'
+       wikis = scope.all # if the user is an admin, show them all the wikis
```

Comment:
@bjblock - I don't know if it's because it's a scope or what but I can't seem to figure out why and how I'm getting `undefined method error` on `role`. I understand that it's because we don't have access to `role` within the Wiki Policy but for some reason, I can't figure out where/how to get role to work within this scope, inside the policy. It seems so easy but I've been struggling with it for an hour or two with no luck. Could you give me a hint?